lib/bugsnag.rb

```diff
@@ -33,12 +33,37 @@ def configure
     end
 
     # Explicitly notify of an exception
-    def notify(exception, auto_notify=false, &block)
-      if auto_notify && !configuration.auto_notify
+    def notify(exception, &block)
+      report = Report.new(exception, configuration)
+      send_report(report, false, &block)
+    end
+
+    # For automatic notification of exceptions
+    def auto_notify(exception, severity_reason, &block)
```

Comment:
I dont think we need the auto_notify method. If you look where we build and send the report blocks when auto_notify is set to true are run first so we can change the reports in the block for auto notifications.

Reply:
So I would just call notify like we did before, just telling notify that this is an auto_notify. Then in the block you can set the severity reason and the severity. Its weird that severity is set in the block, and severity reason outside it right now.